bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/service.js

```diff
@@ -39,36 +39,41 @@ const getPresentations = () =>
       conversion: presentation.conversion || { done: true, error: false },
     }));
 
-const observePresentationConversion = (meetingId, filename) => new Promise((resolve, reject) => {
-  const conversionTimeout = setTimeout(() => {
-    reject({
-      filename,
-      message: 'Conversion timeout.',
-    });
-  }, CONVERSION_TIMEOUT);
+const observePresentationConversion = (meetingId, filename, onConversion) =>
+  new Promise((resolve) => {
+    const conversionTimeout = setTimeout(() => {
+      onConversion({
+        done: true,
+        error: true,
+        status: 'TIMEOUT',
+      });
+    }, CONVERSION_TIMEOUT);
+
+    const didValidate = (doc) => {
+      clearTimeout(conversionTimeout);
+      resolve(doc);
+    };
+
+    Tracker.autorun((c) => {
+    /* FIXME: With two presentations with the same name this will not work as expected */
```

Comment:
You seem to have handled the same-name case. Probably just forgot to remove the FIXME?